### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by czarekk215**

### CODE REVIEW

File: board/e5_dev_board/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+common:
+  build_only: true
+  integration_platforms:
+    - custom_plank
+    - nucleo_f302r8
+tests:
+  app.default: {}
+  app.debug:
+    extra_overlay_confs:
+      - debug.conf
```
Comment (adamkrawczyk) [inline board/e5_dev_board/sample.yaml]: Is this file used for something? if so the platforms are missing e5 board if not needed remove

File: board/e5_dev_board/src/main.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (adamkrawczyk) [inline board/e5_dev_board/src/main.c]: Change this to 
```
/*
 * Copyright (c) WiseVision 2023. All rights reserved.
 */

File: board/e5_dev_board/src/main.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	LOG_INF("Received data: %s (RSSI:%ddBm, SNR:%ddBm)", data, rssi, snr);
+}
+
+void send_msg(const struct device *dev, uint8_t *data, int16_t rssi, int8_t snr){
+	char data_send[MAX_DATA_LEN] = {0};
+	int ret;
+	LOG_INF("Sending data");
+	int convertedValue = atoi(data);
```
Comment (adamkrawczyk) [inline board/e5_dev_board/src/main.c]: please use the same naming rule for variables most of the zephyr projects follow the camel_case rule, change the `convertedValue` to converted_value`

File: board/e5_dev_board/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#-------------------------------------------------------------------------------
+# Zephyr Example Application
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
```
Comment (adamkrawczyk) [inline board/e5_dev_board/CMakeLists.txt]: change license to 
```
/*
 * Copyright (c) WiseVision 2023. All rights reserved.
 */
 ```

File: board/e5_dev_board/Kconfig [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is the application Kconfig entry point. All application Kconfig
+# options can be defined here or included via other application Kconfig files.
+# You can browse these options using the west targets menuconfig (terminal) or
+# guiconfig (GUI).
```
Comment (adamkrawczyk) [inline board/e5_dev_board/Kconfig]: if the file was modified change this as well

File: board/e5_dev_board/prj.conf [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file contains selected Kconfig options for the application.
```
Comment (adamkrawczyk) [inline board/e5_dev_board/prj.conf]: same

File: board/e5_mini/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#-------------------------------------------------------------------------------
+# Zephyr Example Application
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
```
Comment (adamkrawczyk) [inline board/e5_mini/CMakeLists.txt]: .

File: board/e5_mini/sample.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+common:
+  build_only: true
+  integration_platforms:
+    - custom_plank
+    - nucleo_f302r8
+tests:
+  app.default: {}
+  app.debug:
+    extra_overlay_confs:
+      - debug.conf
```
Comment (adamkrawczyk) [inline board/e5_mini/sample.yaml]: is this file needed?

File: board/e5_mini/src/main.c [hunk 9]
```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_mini/src/main.c [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+
+#define MAX_DATA_LEN 10
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lora_send);
+
+int map(int value, int inMin, int inMax, int outMin, int outMax) {
+    return (value - inMin) * (outMax - outMin) / (inMax - inMin) + outMin;
+}
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: change the function name to something that explains more what this function does.

File: board/e5_mini/src/main.c [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+
+// configure transmission
+int configureTx(const struct device *const lora_dev)
+{
+	struct lora_modem_config config;
+	config.frequency = 865100000;
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c:83]: maybe add a comment that this is eu freqz and what should be for the us

File: board/e5_mini/src/main.c [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+	}
+	return ret;
+}
+
+void send_msg(const struct device *dev, int moisture){
+	int ret;
+	printk("Received moisture value: %d\n", moisture);
+	char data[MAX_DATA_LEN];
+	char moisture_str[MAX_DATA_LEN];
+	snprintf(moisture_str, sizeof(moisture_str), "%.2d", moisture);
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: if possible please use one way of logging. Possibly the zephyr ways of logging `LOG_INF....`

File: board/e5_mini/src/main.c [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+	memset(data, 0, sizeof(data));
+	strncpy(data, moisture_str, sizeof(data) - 1);
+	/*for (int i = 0; i < strlen(moisture_str) && i < MAX_DATA_LEN; ++i) {
+	    data[i] = moisture_str[i];
+	}*/
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: remove comment

File: board/e5_mini/src/main.c [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+	}
+
+
+}
+
+void adc_config(){
+	int err;
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+		if (!adc_is_ready_dt(&adc_channels[i])) {
+			printk("ADC controller device %s not ready\n", adc_channels[i].dev->name);
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_mini/src/main.c [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+			return 0;
+		}
+
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			printk("Could not setup channel #%d (%d)\n", i, err);
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_mini/src/main.c [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+		int val_int = (int)val_mv;
+		moisture = map(val_int, DryValue, WetValue, 0, 100);
+
+	}
+	return moisture;
+}
+
+void set_led(uint8_t *data_recive, const struct device *gpio1, const struct device *gpio2){
+	int ret;
+	int SetLed = atoi(data_recive);
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c:217]: camel_cased?

File: docs/asset/.gitkeep [hunk 17]
```diff
@@ -0,0 +1 @@
+
```
Comment (adamkrawczyk) [inline docs/asset/.gitkeep]: remove this file

**Review by adamkrawczyk — CHANGES_REQUESTED** [review]

File: board/e5_dev_board/sample.yaml (hunk 1 above)
Comment (czarekk215) [reply board/e5_dev_board/sample.yaml]: Removed

File: README.md [hunk 18]
```diff
@@ -1 +1,415 @@
-# project-cezary
\ No newline at end of file
+# Soil moisture system using grove capicty moisture sensor, Wio E5 dev board and Wio E5 mini board
+## Introduction
+![System schema](./docs/asset/project_cezary_1.drawio.png)
```
Comment (styczen) [inline README.md:4]: Add addional line after because new line is appended to image
image

File: README.md [hunk 19]
```diff
@@ -1 +1,415 @@
...
+* Reading soil mositure level via ADC on mini board.
+* Sending data about mositure level via Lora to dev board.
+* Dev board recive this data and send order to mini board about, what color rgb led, should lights.
+## Technologies
+Project is created with:
+* Zephyr: 3.5.0
+* West: 1.2.0
+## Preparation
+1. It is necessary to have Zephyr already installed to build or flash application.
+   * Go to the official Zephyr RTOS website (https://www.zephyrproject.org/) to download the latest version.
```
Comment (styczen) [inline README.md]: I would add more direct link to download page. When I opened this link, there is no "Download" button visible.

File: README.md [hunk 20]
```diff
@@ -1 +1,415 @@
-# project-cezary
\ No newline at end of file
+# Soil moisture system using grove capicty moisture sensor, Wio E5 dev board and Wio E5 mini board
+## Introduction
+![System schema](./docs/asset/project_cezary_1.drawio.png)
+Soil moisture system includes:
+* Reading soil mositure level via ADC on mini board.
+* Sending data about mositure level via Lora to dev board.
```
Comment (styczen) [inline README.md]: Nit:
```suggestion
* Sending data about mositure level via LoRa to dev board.
```

File: README.md [hunk 21]
```diff
@@ -1 +1,415 @@
...
+* Dev board recive this data and send order to mini board about, what color rgb led, should lights.
```
Comment (styczen) [inline README.md]:
```suggestion
* Dev board receive this data and send order to mini board about, what color rgb led, should lights.
```

File: README.md [hunk 22]
```diff
@@ -1 +1,415 @@
...
+west build -b lora_e5_dev_board
+```
+Successful build should looks like this:
+![Successful build](./docs/correct_build.png)
+4. Upload your build to board:
+```
+west flash
+```
+Successful flash should looks like this:
+![Successful flash](./docs/asset/correct_flash.png)
```
Comment (styczen) [inline README.md]: Maybe it might be good to add some information about connecting board to computer i.e. if some programmer/debugger is needed or just standard USB cable (and what type: C, micro, mini etc). Probably update graphics from the beginning.

File: README.md [hunk 23]
```diff
@@ -1 +1,415 @@
...
+	config.tx = false;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+```
+This function is for configuarting reception. Lora newtork has few things to confing, but we have only to choose the frequnecy. Frequnecy is diffrent for diffrent regions.
```
Comment (styczen) [inline README.md]:
```suggestion
This function is for receiving configuration. LoRa network has a few things to configure, but we have to choose only the frequnecy. Frequnecy is different for different regions.
```

File: README.md [hunk 24]
```diff
@@ -1 +1,415 @@
...
+	config.tx = true;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+```
+This function is for configuarting transmission. Like in previous function we have to choose to correct frequnecy.
```
Comment (styczen) [inline README.md]:
```suggestion
This function is for transmission configuration. Like in previous function, we have to choose to correct frequnecy.
```

File: README.md [hunk 25]
```diff
@@ -1 +1,415 @@
...
+	ret=lora_send(dev,data_send,1);
+	if (ret < 0) {
+		LOG_ERR("LoRa send failed");
+	}else{
+		LOG_INF("Data send:%s",data_send);
+	}
+
+}
+```
+This function send data 0 if the soil moisture is accurate and 1 if our plant needs water. The correct level of moisture is set for under 30 %. It could be change for your purpose.
```
Comment (styczen) [inline README.md]:
```suggestion
This function send data 0 if the soil moisture is correct and 1 if our plant needs water. The correct level of moisture is set for under 30 %. It could be change for your purpose.
```
I assume that you meant correct in this case i.e. soil moisture is at the correct level.

File: README.md [hunk 26]
```diff
@@ -1 +1,415 @@
...
+	while(1){
+		recev_msg(lora_dev, data, rssi, snr);
+		send_msg(lora_dev, data, rssi, snr);
+		k_sleep(K_MSEC(1000));
+	}
+	k_sleep(K_FOREVER);
+	return 0;
+}
+```
+In main function we recev and send messegs form board to bord with 1 second interval.
```
Comment (styczen) [inline README.md]:
```suggestion
In main function we receive and send messegs form board to board with 1 second interval.
```

File: README.md [hunk 27]
```diff
@@ -1 +1,415 @@
...
+`configureRx and configureTx`
+This functions are the same as on dev boards. If we want communication between two boards, we have to set the same frequency.
+```c
+const int DryValue = 1950;
+const int WetValue = 1220;
+int mapSoilMoisture(int value, int inMin, int inMax, int outMin, int outMax) {
+    return (value - inMin) * (outMax - outMin) / (inMax - inMin) + outMin;
+}
+```
+This function is used to translet masurments from ADC to 0-100 values. If we want to have correct values of moisture, we have to make a calibration of the sensor.
```
Comment (styczen) [inline README.md]:
```suggestion
This function is used to translate measurement from ADC to 0-100 values. If we want to have correct values of moisture, we have to calibrate the sensor. 
```

File: README.md [hunk 28]
```diff
@@ -1 +1,415 @@
...
+	len = lora_recv(dev, data_recive, MAX_DATA_LEN, K_FOREVER, &rssi, &snr);
+	if (len < 0) {
+		LOG_ERR("LoRa receive failed");
+		return 0;
+	}
+
+	LOG_INF("Received data: %s", data_recive);
+}
+```
+This function recevs data from dev board.
```
Comment (styczen) [inline README.md]:
```suggestion
This function receives data from dev board.
```

File: README.md [hunk 29]
```diff
@@ -1 +1,415 @@
...
+		return;
+	}
+	if(!device_is_ready(gpio2)){
+		return;
+	}
+
+
+}
+```
+This function configs gpios outputs for led controling.
```
Comment (styczen) [inline README.md]:
```suggestion
This function configures gpios outputs for led controlling.
```

File: README.md [hunk 30]
```diff
@@ -1 +1,415 @@
...
+
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			LOG_INF("Could not setup channel #%d (%d)\n", i, err);
+			return 0;
+		}
+	}
+}
+```
+This function is used to configs analog to digital converter.
```
Comment (styczen) [inline README.md]:
```suggestion
This function is used to configure analog to digital converter.
```

File: README.md [hunk 31]
```diff
@@ -1 +1,415 @@
...
+```
+This function sets color of leds depending on the values send from dev board.
+## Finishing
+If you flashed your board with built programms you can check their logs.
+To see logs
+1. Open terminal
+2. Use command `sudo picocom --baud 115200 /dev/"your_usb_port"`
+### On Wio E5 dev board kit logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_dev_board.png)
+* Recive data informs us that data has been recived
```
Comment (styczen) [inline README.md]:
```suggestion
* Receive data informs us that data has been received
```

File: README.md [hunk 32]
```diff
@@ -1 +1,415 @@
...
+* Recived data shows value of moisture
```
Comment (styczen) [inline README.md:422]:
```suggestion
* Received data shows value of moisture
```

File: README.md [hunk 33]
```diff
@@ -1 +1,415 @@
...
+* Sending data infroms us that data has been sended
```
Comment (styczen) [inline README.md]:
```suggestion
* Sending data informs us that data has been sended
```

File: README.md [hunk 34]
```diff
@@ -1 +1,415 @@
...
+* Data send shows value that has been send: 1 means you need too water the plant and 0 means that your soil moisture is correct
+* Blinking led on board informs us that there is connection via uart.
+### On Wio E5 mini board logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_mini_board.png)
+* Data send! infroms us that data has been sended
+* Recive data informs us that data has been recived
+* Recive data shows value sended from dev board: 1 means you need too water the plant and 0 means that your soil moisture is correct
```
Comment (styczen) [inline README.md]:
```suggestion
* Receive data informs us that data has been received
* Receive data shows value sended from dev board: 1 means you need too water the plant and 0 means that your soil moisture is correct
```

**Review by styczen — CHANGES_REQUESTED** [review]
README in general is clear. I found few typos.

I just skimmed the code and one thing I would recommend is to use some formatter.  For C/C++ there is [clang-format](https://clang.llvm.org/docs/ClangFormat.html).

If you do not want to format code by hand, you can use [Git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) e.g. `pre-commit` or use your editor of choice to format code when you save.

Other code style thing is to choose one style and stick to it. I saw in a few places `camelCase` and in the other `lower_case_with_underscores`

**Comment by domrobotec** [comment]
Some general remarks.

When you are working on a particular branch just yourself please use more often: `git commit --amend --no-edit`. You can avoid multiple commits with the same message. Right now there are 48 commits from which more than half have "Update README" message :smile: 
It can sometimes require force pushing, but don't be afraid of it when you are working on your branch. Try not to force push when there are more people working on the same branch.

File: README.md [hunk 35]
```diff
@@ -1 +1,415 @@
...
+![System schema](./docs/asset/project_cezary_1.drawio.png)
+Soil moisture system includes:
+* Reading soil mositure level via ADC on mini board.
+* Sending data about mositure level via Lora to dev board.
+* Dev board recive this data and send order to mini board about, what color rgb led, should lights.
+## Technologies
+Project is created with:
+* Zephyr: 3.5.0
+* West: 1.2.0
+## Preparation
```
Comment (domrobotec) [inline README.md]: Prerequisites

File: README.md [hunk 36]
```diff
@@ -1 +1,415 @@
...
+2. Install Build Tools:
+* Zephyr RTOS uses build tools like CMake and ninja. You have to follow this commmands to install:
+* CMake:
+```
+sudo apt-get update
+sudo apt-get install cmake
+```
+* Ninja:
+```
+sudo apt-get update
```
Comment (domrobotec) [inline README.md]: As this is one preparation process, you don't have to add `sudo apt-get update` for each build tool. I would leave the first one only.

File: README.md [hunk 37]
```diff
@@ -1 +1,415 @@
...
+## Technologies
+Project is created with:
+* Zephyr: 3.5.0
+* West: 1.2.0
+## Preparation
+1. It is necessary to have Zephyr already installed to build or flash application.
+   * Go to the official Zephyr RTOS website (https://www.zephyrproject.org/) to download the latest version.
+   * Follow the installation instructions on the website to install Zephyr RTOS on your system.
+2. Install Build Tools:
+* Zephyr RTOS uses build tools like CMake and ninja. You have to follow this commmands to install:
```
Comment (domrobotec) [inline README.md]: I don't like the phrase "You have to follow this commands".
I would change the whole part to:

Zephyr RTOS uses build tools like CMake and Ninja:
```
sudo apt-get update
sudo apt-get install cmake
sudo apt-get install ninja-build
```

If you want to you can change the first line to something like:
Zephyr RTOS uses build tools like CMake and Ninja. Use following commands to install both:

File: README.md (hunk 18 above)
Comment (domrobotec) [reply README.md:4]: Also, use the same grammatical structure for all of the bullet points. e.g.:
The soil moisture system includes:
- Reading the soil moisture level via ADC on the mini board.
- Transmitting data about the moisture level via LoRa to the dev board.
- Receiving this data by the dev board and sending instructions to the mini board about which color the RGB LED should light up.

Additionally, in the image above, I assume that the left board is the mini board, but it is not 100% clear. You can add some explanation at the very beginning, e.g., "The system consists of the following hardware: ..." 
Or you can just combine all of these information into one list :smiley:

File: README.md [hunk 38]
```diff
@@ -1 +1,415 @@
...
+* Sending data about mositure level via Lora to dev board.
+* Dev board recive this data and send order to mini board about, what color rgb led, should lights.
+## Technologies
+Project is created with:
+* Zephyr: 3.5.0
+* West: 1.2.0
+## Preparation
+1. It is necessary to have Zephyr already installed to build or flash application.
+   * Go to the official Zephyr RTOS website (https://www.zephyrproject.org/) to download the latest version.
+   * Follow the installation instructions on the website to install Zephyr RTOS on your system.
```
Comment (domrobotec) [inline README.md]: If we redirect somebody to another website, I would also give a link, so that you don't have to look for the right instructions.

File: README.md [hunk 39]
```diff
@@ -1 +1,415 @@
...
+sudo apt-get install ninja-build
+```
+3. Install picom
+* To communicate with your board you have to install picom:
+```
+sudo apt-get update
+sudo apt-get -y install picom
+```                                                         
+## E5 dev board setup             
+To build versions for the e5 dev board, you need an unlocked board(this shows how to unlock board https://wisevision.tech/docs/LoRa/LoRa-e5-dev-board-unprotect-memory), connected via UART and st-link.
```
Comment (domrobotec) [inline README.md]: What are "versions" here?

Space before opening the brackets, no comma after brackets.

"connected via UART and st-link."
I am not an expert in the field, but is it obvious to what the board should be connected?  PC in this case?

File: README.md [hunk 40]
```diff
@@ -1 +1,415 @@
...
+1. You need to copy Source code from github into zephyrpoject directory:
```
Comment (domrobotec) [inline README.md]: Don't start every instruction with "you need". Simple: "Copy the source code from github into `/zephyrpoject` directory" is enough.

File: README.md [hunk 41]
```diff
@@ -1 +1,415 @@
...
+```
+cd /zephyrpoject
+git clone https://github.com/wise-vision/project-cezary.git
+```
+2. You need to set the path to the code file:
```
Comment (domrobotec) [inline README.md]: `cd` is used to change directory. 
Maybe something like:
Navigate to the build directory:
or
Go the the build directory:

Change the "build directory" to whatever this directory refers to :smiley:

File: README.md (hunk 22 above)
Comment (domrobotec) [inline README.md]: I don't like the idea of attaching screenshots of terminal output.
I can see that the first image corresponds to build process. Maybe text description like this can replace the image:
Build is successful when all of the build jobs are finished. To test, upload the build to the board.

Regarding the second image, I think @adamkrawczyk is a much more experienced person to review.

File: README.md [hunk 42]
```diff
@@ -1 +1,415 @@
...
+```
+Successful build should looks like this:
+![Successful build](./docs/correct_build.png)
+4. Upload your build to board:
+```
+west flash
+```
+Successful flash should looks like this:
+![Successful flash](./docs/asset/correct_flash.png)
+5. After build and flash to the board, to run the app you have to press the reset button on board.
```
Comment (domrobotec) [inline README.md]: remove "you have to"

File: README.md [hunk 43]
```diff
@@ -1 +1,415 @@
...
+![Wio E5 dev board reset button](./docs/asset/reset_dev_board.png)
+## E5 mini board setup
+To build versions for the e5 mini board, you need an unlocked board(this shows how to unlock board https://wisevision.tech/docs/LoRa/LoRa-e5-dev-board-unprotect-memory), connected via UART and st-link.
```
Comment (domrobotec) [inline README.md:64]: ditto

File: README.md [hunk 44]
```diff
@@ -1 +1,415 @@
...
+1. You need to copy Source code from github into zephyrpoject directory:
```
Comment (domrobotec) [inline README.md]: ditto for whole section

File: README.md [hunk 45]
```diff
@@ -1 +1,415 @@
...
+## Code of E5 dev board
+### prj.conf
+```c
+CONFIG_LOG=y
+CONFIG_SPI=y
+CONFIG_GPIO=y
+CONFIG_LORA=y
+CONFIG_CBPRINTF_FP_SUPPORT=y
+```
+These configurations help customize the Zephyr OS for specific hardware and functionality requirements.
```
Comment (domrobotec) [inline README.md]: Place explanation before the code.

File: README.md (hunk 23 above)
Comment (domrobotec) [reply README.md]: ditto

File: README.md [hunk 46]
```diff
@@ -1 +1,415 @@
...
+![Successful build](./docs/correct_build.png)
+4. Upload your build to board:
+```
+west flash
+```
+Successful flash should looks like this:
+![Successful flash](./docs/asset/correct_flash.png)
+5. After build and flash to the board, to run the app you have to press the reset button on board.
+![Wio E5 mni board reset button](./docs/asset/reset_mini_board.png)
+## Code of E5 dev board
```
Comment (domrobotec) [inline README.md:91]: For the whole section, please refactor. First shortly explain what the piece of code does, next place the piece of code, and next add some more details about the code if needed.

File: README.md [hunk 47]
```diff
@@ -1 +1,415 @@
...
+		}
+		
+		ret = gpio_pin_set_raw(gpio1, 0, 0);
+		ret = gpio_pin_set_raw(gpio2, 10, 1);
+			
+	} 
+}
+```
+This function sets color of leds depending on the values send from dev board.
+## Finishing
```
Comment (domrobotec) [inline README.md]: Add newline before each section

File: README.md [hunk 48]
```diff
@@ -1 +1,415 @@
...
+If you flashed your board with built programms you can check their logs.
```
Comment (domrobotec) [inline README.md]: programs

File: README.md (hunk 47 above)
Comment (domrobotec) [inline README.md]: I don't like "Finishing" as the name of the section but I don't have any idea how to name it. @styczen @adamkrawczyk

File: README.md [hunk 49]
```diff
@@ -1 +1,415 @@
...
+2. Use command `sudo picocom --baud 115200 /dev/"your_usb_port"`
+### On Wio E5 dev board kit logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_dev_board.png)
+* Recive data informs us that data has been recived
+* Recived data shows value of moisture
+* Sending data infroms us that data has been sended
+* Data send shows value that has been send: 1 means you need too water the plant and 0 means that your soil moisture is correct
+* Blinking led on board informs us that there is connection via uart.
+### On Wio E5 mini board logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_mini_board.png)
```
Comment (domrobotec) [inline README.md]: This is why I don't like pictures of text in the README. The images are not proportional and the size of letters varies a lot. Use:
```sh
blabla 
goo
foo
```

File: README.md [hunk 50]
```diff
@@ -1 +1,415 @@
...
+* Data send! infroms us that data has been sended
```
Comment (domrobotec) [inline README.md]: Try to be more careful with grammar :) Also, when you can, try to avoid making the instruction personal. In most cases you don't need to use words like: "us", "you", "your".
"Data Sent!" informs that the data has been sent.
"Receive Data" informs that the data has been received.
"Receive Data" shows the value sent from the dev board: 1 indicates a need for watering the plant, and 0 indicates that the soil moisture is correct.

Comment (domrobotec) [reply README.md]: Change the whole section accordingly

**Comment by domrobotec** [comment]
In general the instructions were clear for me with some minor exceptions :)

File: board/e5_dev_board/sample.yaml (hunk 1 above)
Comment (adamkrawczyk) [reply board/e5_dev_board/sample.yaml]: if this is resolved then click resolve conversation

**Comment by adamkrawczyk** [comment]
> Some general remarks.
> 
> When you are working on a particular branch just yourself please use more often: `git commit --amend --no-edit`. You can avoid multiple commits with the same message. Right now there are 48 commits from which more than half have "Update README" message 😄 It can sometimes require force pushing, but don't be afraid of it when you are working on your branch. Try not to force push when there are more people working on the same branch.

i guess it was due to edit on github

File: README.md (hunk 47 above)
Comment (adamkrawczyk) [reply README.md]: Maybe `Conclusion` ?

File: README.md (hunk 49 above)
Comment (adamkrawczyk) [reply README.md]: Add extra line here

File: board/e5_mini/src/main.c [hunk 51]
```diff
@@ -0,0 +1,245 @@
...
+	config.tx = true;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+
+void send_msg(const struct device *dev, int moisture){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: keep all fuc names in the same convention use camelCased

File: board/e5_mini/src/main.c [hunk 52]
```diff
@@ -0,0 +1,245 @@
...
+	ret = lora_send(dev, data, MAX_DATA_LEN);
+	if (ret < 0) {
+		LOG_ERR("LoRa send failed");
+		return 0;
+	}
+
+	LOG_INF("Data sent!");
+
+}
+void recev_msg(const struct device *dev, uint8_t *data_recive, int16_t rssi, int8_t snr){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: camelCased

File: board/e5_mini/src/main.c [hunk 53]
```diff
@@ -0,0 +1,245 @@
...
+	len = lora_recv(dev, data_recive, MAX_DATA_LEN, K_FOREVER, &rssi, &snr);
+	if (len < 0) {
+		LOG_ERR("LoRa receive failed");
+		return 0;
+	}
+
+	LOG_INF("Received data: %s", data_recive);
+}
+
+void gpios_config(const struct device *gpio1, const struct device *gpio2){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_mini/src/main.c [hunk 54]
```diff
@@ -0,0 +1,245 @@
...
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			LOG_INF("Could not setup channel #%d (%d)\n", i, err);
+			return 0;
+		}
+	}
+}
+const int DryValue = 1950;
+const int WetValue = 1220;
+int adc_val(){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: camelCased

File: board/e5_mini/src/main.c [hunk 55]
```diff
@@ -0,0 +1,245 @@
...
+		}
+		err = adc_raw_to_millivolts_dt(&adc_channels[i],&val_mv);
+		int val_int = (int)val_mv;
+		moisture = mapSoilMoisture(val_int, DryValue, WetValue, 0, 100);
+
+	}
+	return moisture;
+}
+
+void set_led(uint8_t *data_recive, const struct device *gpio1, const struct device *gpio2){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_mini/src/main.c [hunk 56]
```diff
@@ -0,0 +1,245 @@
...
+		return;
+	}
+	if(!device_is_ready(gpio2)){
+		return;
+	}
+
+
+}
+
+void adc_config(){
```
Comment (adamkrawczyk) [inline board/e5_mini/src/main.c]: .

File: board/e5_dev_board/src/main.c [hunk 57]
```diff
@@ -0,0 +1,115 @@
...
+	len = lora_recv(dev, data, MAX_DATA_LEN, K_FOREVER, &rssi, &snr);
+	if (len < 0) {
+		LOG_ERR("LoRa receive failed");
+		return 0;
+	}
+
+	LOG_INF("Received data: %s (RSSI:%ddBm, SNR:%ddBm)", data, rssi, snr);
+}
+
+void send_msg(const struct device *dev, uint8_t *data, int16_t rssi, int8_t snr){
```
Comment (adamkrawczyk) [inline board/e5_dev_board/src/main.c]: camelCased

File: board/e5_dev_board/src/main.c [hunk 58]
```diff
@@ -0,0 +1,115 @@
...
+	config.tx = true;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+
+void recev_msg(const struct device *dev, uint8_t *data, int16_t rssi, int8_t snr){
```
Comment (adamkrawczyk) [inline board/e5_dev_board/src/main.c]: camelCased

**Review by adamkrawczyk — CHANGES_REQUESTED** [review]

File: board/e5_mini/src/main.c (hunk 51 above)
Comment (domrobotec) [reply board/e5_mini/src/main.c]: https://stackoverflow.com/a/1722518

Comment (domrobotec) [reply board/e5_mini/src/main.c]: https://www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference

Comment (domrobotec) [reply board/e5_mini/src/main.c]: I am confused. I assume, whatever we use is fine as long as we are consistent with one convention

File: board/e5_dev_board/Kconfig (hunk 5 above)
Comment (domrobotec) [reply board/e5_dev_board/Kconfig]: Have you researched the topic of changing license? I know, that Apache-2.0 provides the code AS IS, but I don't have broad knowledge about licensing. I just want to be safe in this regard, there is nothing wrong with using open source code in the product.

**Review by adamkrawczyk — APPROVED** [review]
lgtm
